compiler/src/dotty/tools/dotc/typer/Inferencing.scala

```diff
@@ -153,41 +153,22 @@ object Inferencing {
   def isSkolemFree(tp: Type)(implicit ctx: Context): Boolean =
     !tp.existsPart(_.isInstanceOf[SkolemType])
 
-  /** Derive information about a pattern type by comparing it with some variant of the
-   *  static scrutinee type. We have the following situation in case of a (dynamic) pattern match:
+  /** Infer constraints that should be in scope for a case body with given pattern and scrutinee types.
    *
-   *       StaticScrutineeType           PatternType
-   *                         \            /
-   *                      DynamicScrutineeType
```

Comment:
I think this diagram is useful info to have. Is it wrong? If not let's keep it.

Reply:
Restored the old comment.